example/CMakeLists.txt

```diff
@@ -49,11 +49,11 @@ foreach(file ${files})
         	${Boost_LIBRARIES} ${LIBEVENT_LIBRARIES} ${JSONCPP_LIBRARIES})
         endif()
     else()
-        #if (BUILD_ROCKETMQ_SHARED)
+        if (BUILD_ROCKETMQ_SHARED)
     	    target_link_libraries (${basename} rocketmq_shared ${deplibs})
-        #else()
-    	#    target_link_libraries (${basename} rocketmq_static ${deplibs})
-        #endif()
+        else()
+    	    target_link_libraries (${basename} rocketmq_static ${deplibs} ${Boost_LIBRARIES} ${LIBEVENT_LIBRARIES} ${JSONCPP_LIBRARIES})
```

Comment:
I think boost libraries, libevent, and jsonncpp library don't need to be linked again, as they are linked to rocketmq_static before. 
Could you please remove "${Boost_LIBRARIES} ${LIBEVENT_LIBRARIES} ${JSONCPP_LIBRARIES}" and have a test on it?